syntax = "proto3";

package pypoly.pypet;

message ContextVar {
  enum VarType {
    INT32 = 0;
    INT32_ARRAY = 1;
    TENSOR_ARRAY = 2;
  };

  string name = 1;   // name of the variable in the codes.
  VarType type = 2;  // type of the variable.
  repeated int32 lower_bound = 3;
  repeated int32 upper_bound = 4;

  message ElementDesc {
    string elem_type = 1;  // "int" or "tensor";
    repeated int64 shape = 2;
  };
  ElementDesc elem_desc = 5;
}

message ContextDesc {
  repeated ContextVar vars = 1;
}
